src/components/shared/Entity/Create.tsx

```diff
@@ -96,6 +121,38 @@ function EntityCreate({
         setDraftId(null);
     }, [imageTag, setDraftId]);
 
+    const { connectorTag } = useConnectorTag(imageTag.id);
+    const { liveSpecs } = useLiveSpecsExtWithOutSpec(specId, ENTITY.CAPTURE);
+    const { liveSpecs: liveSpecsByLastPub } = useLiveSpecsExtByLastPubId(
+        lastPubId,
+        ENTITY.CAPTURE
+    );
+
+    useEffect(() => {
+        if (connectorTag) {
+            setEndpointSchema(connectorTag.endpoint_spec_schema);
+            setResourceSchema(connectorTag.resource_spec_schema);
+            // We wanna make sure we do these after the schemas are set as
+            //  as they are dependent on them.
+            if (liveSpecs.length > 0) {
+                prefillCollections(liveSpecs);
+            } else if (liveSpecsByLastPub.length > 0) {
+                prefillCollections(liveSpecsByLastPub);
+            }
+        }
+    }, [
+        connectorTag,
+        liveSpecs,
+        liveSpecsByLastPub,
+        prefillCollections,
+        setEndpointSchema,
+        setResourceSchema,
+    ]);
```

Comment:
This is basically a big "controller" for setting stuff into the create store. Not super into this pattern but it works okay for now.